Wayland 1.1 is enforcing the protocol whereas 1.0 was lenient.  Backport a patch
from git to fix repainting.

Upstream-Status: Backport
Signed-off-by: Ross Burton <ross.burton@intel.com>


From 0d2c4617203c8fe907c722c9cb53c0345e0405e7 Mon Sep 17 00:00:00 2001
From: Scott Moreau <oreaus@gmail.com>
Date: Sun, 10 Mar 2013 10:51:01 -0600
Subject: [PATCH] wayland: Always attach buffer before committing

With recent changes in attach semantics, we always need to attach before
committing. Without this changes to the window contents to not get reflected
in the content of the surface.

Signed-off-by: Rob Bradford <rob@linux.intel.com>
---
 gdk/wayland/gdkwindow-wayland.c |    7 -------
 1 file changed, 7 deletions(-)

diff --git a/gdk/wayland/gdkwindow-wayland.c b/gdk/wayland/gdkwindow-wayland.c
index 112a1f4..fa01e90 100644
--- a/gdk/wayland/gdkwindow-wayland.c
+++ b/gdk/wayland/gdkwindow-wayland.c
@@ -327,13 +327,6 @@ gdk_wayland_window_attach_image (GdkWindow *window)
   if (GDK_WINDOW_DESTROYED (window))
     return;
 
-  /* The "drawn to" Cairo surface is the same as the Cairo surface from which
-   * we are driving the buffer for the Wayland surface. Therefore we don't
-   * need to do anything here
-   */
-  if (impl->server_surface == impl->cairo_surface)
-    return;
-
   /* The wayland surface is attached to a buffer that is from the old "drawn
    * to" surface. Unref the surface and restore the state.
    */
-- 
1.7.10.4

